openstacksdk_plugin/decorators.py

```diff
@@ -57,15 +61,42 @@ def get_property_by_name(property_name):
 
             client_config = get_property_by_name('client_config')
             resource_config = get_property_by_name('resource_config')
-            resource_id = get_property_by_name(RESOURCE_ID)
-            if resource_id:
-                resource_config['id'] = resource_id
+            resource = class_decl(client_config=client_config,
+                                  resource_config=resource_config,
+                                  logger=ctx.logger)
+
+            # # Check if "use_external_resource" is set to True
+            is_external = get_property_by_name(USE_EXTERNAL_RESOURCE_PROPERTY)
+            if is_external and operation_name in ['create', 'delete']:
+                if not resource_config.get(RESOURCE_ID):
```

Comment:
you can just do `resource.get()`, because the `class_decl` loads the `resource_config` into a class attribute named `config`. The resource `.get` function then implements whether to use `self.name` or `self.id` in the `.get` function.

Reply:
Yup you are right will update it